ibis/expr/types/arrays.py

```diff
@@ -1058,18 +1084,21 @@ def __getitem__(self, index: int | ir.IntegerValue | slice) -> ir.Column:
 def array(values: Iterable[V]) -> ArrayValue:
     """Create an array expression.
 
+    If any values are Columns, the result will be an ArrayColumn.
+    Otherwise, the result will be an ArrayScalar.
```

Comment:
I think that the 3 different datashapes of Scalar, Column, and Table are essential for a user to understand. A FloatingColumn has a .mean() function, but a FloatingScalar doesn't. When I .execute() something, am I going to get a python `int`, or pandas series of ints? I need to know which shape I am working with. I think I would have been a lot less confused with this when I first picked up ibis if this had been more forward.

The datashapes uses the same concept of broadcasting from SQL and from numpy, so for experienced users it shouldn't be new. For newbies, it might be confusing at first, but still I think it is essential to understand, and more of a disservice to them to hide it.

IDK this might just be an opinion deal though, so maybe we should get more user input here?

Reply:
I would love it if you wanted to take the lead on that.

Reply:
See https://github.com/ibis-project/ibis/pull/8557

Reply:
Should I add a link to that documentation page? If so, I'm not sure what the syntax is...

Reply:
Should be the same as a regular markdown link, with an absolute path starting after `docs`

Reply:
Done.